### ISSUE TITLE
I dont know the difference from GAN.

### ISSUE THREAD

**Issue by AlbertLin0** [mlvlab/DAVI#1]
<img width="375" alt="image" src="https://github.com/user-attachments/assets/1dd550d8-e712-4ba9-929b-e71239dd08e4">

My understanding: the picture above has a network $I_\phi $, which is the author's aim to play a  role of one-shot inference. The network is trained with variational method and the lower bound is the picture follow.

<img width="590" alt="image" src="https://github.com/user-attachments/assets/151be32e-f6ee-4d43-af48-91b2d067b2a6">

But, if you only take notice to loss function and network $I_\phi$, there is two questions:
- why dont use truth as $p(x)$, perhaps the diffusion outputs Gaussian means as $p(x)$..
- what's the difference from GAN? freeze the Discriminator to train generartor...

---
THANKS

**Comment by sojinleeme** [comment]
Thank you for your attention:)

- Q1: Why not use p(x)?
A1: We cannot explicitly compute $$p(x)$$, so we approximate it using a pre-trained diffusion model. The diffusion model provides the derivative of $$log (p(x))$$.

- Q2: How does this differ from GANs?
A2: Our one-step network resembles GANs in terms of one-step sampling. However, our loss function is fundamentally different from the GAN approach. Unlike GANs, we use the score distillation loss derived from variational inference.

Please refer to our paper for a detailed explanation. 
We have thoroughly described our main objectives from the perspective of variational inference.

**Comment by a-little-hoof** [comment]
Hi, I really like your work and appreciate the contribution of solving inverse problem with amortized inference.

However, while I was reading the source code, I found that apart from the framework shown in the above picture, you also implement `self.criterion = LPIPSWithDiscriminator2D(disc_in_channels=3, disc_weight=1.0)`, which also contributes to the loss. 

As a result, my question is that whether this discriminator is helpful to the framework? If so, why didn't you mention the discriminator in the paper? I might miss a few details, sorry for disturbing you.

**Comment by sojinleeme** [comment]
Dear @a-little-hoof,
Thank you so much for taking an interest in my work.

Inspired by LDM, we have also implemented adversarial loss along with L2 loss in order to see its impact. 
We have observed that it introduces a trade-off between PSNR and LPIPS performance but does not have a significant impact overall. It can be easily turned off by setting its weight to 0.

For this reason, it wasn’t emphasized in the paper, as we aimed to focus on the core aspects of the framework.
